Add docs for the Fabric `WandbLogger` hosted in the `wandb` SDK (#19451)
Lightning-AI/pytorch-lightning: docs/source-fabric/guide/loggers/wandb.rst

.. code-block:: python

    value = ...  # Python scalar or tensor scalar
    fabric.log("some_value", value)

To log multiple metrics at once, use :meth:`~lightning.fabric.fabric.Fabric.log_dict`:

.. code-block:: python

    values = {"loss": loss, "acc": acc, "other": other}
    fabric.log_dict(values)

----

**************************************************
Logging media, artifacts, hyperparameters and more
**************************************************

With ``WandbLogger`` you can also log images, text, tables, checkpoints, hyperparameters and more.
For a description of all features, check out the official Weights and Biases documentation and examples.

.. raw:: html

    <div class="display-card-container">
        <div class="row">

.. displayitem::
    :header: Official WandbLogger Lightning and Fabric Documentation
    :description: Learn about all features from Weights and Biases
    :button_link: https://docs.wandb.ai/guides/integrations/lightning
    :col_css: col-md-4
    :height: 150

.. displayitem::
    :header: Fabric WandbLogger Example
    :description: Official example of how to use the WandbLogger with Fabric

>     :button_link: https://colab.research.google.com/github/wandb/examples/blob/master/colabs/pytorch-lightning/Track_PyTorch_Lightning_with_Fabric_and_Wandb.ipynb

Reviewer: Could it be rather a release branch?

Reviewer 2: This should be fine as a starting point.
I lost contact with @ash0ts so I'm not sure if he's going to see this. But I'll merge it for now and hope the link won't get oudated too quickly.

Author: Hello! I just returned PTO with no access to proper internet. What do you mean by `release branch` in this context?

##################
Weights and Biases
##################

> `Weights & Biases (W&B) <https://wandb.ai>`_ is the AI developer platform, with tools for training models, fine-tuning models, and leveraging foundation models.
> 
> W&B allows machine learning practitioners to track experiments, visualize data, and share insights with a few lines of.

Reviewer 2: We have a small request here:
Since Lightning AI is also "the AI developer platform with tools for training models etc etc" there is a slight conflict here. Could we maybe keep the focus of the intro on the experiment tracking aspect? The second paragraph is perfect, it's just this sentence here that we would like to adjust.

Maybe just merging it like so would work:

```suggestion
`Weights & Biases (W&B) <https://wandb.ai>`_  allows machine learning practitioners to track experiments, visualize data, and share insights with a few lines of code.

```

Otherwise awesome docs!
